src/main/java/com/exalt/resourcemanagementservice/service/ResourceService.java

```diff
@@ -0,0 +1,52 @@
+package com.exalt.resourcemanagementservice.service;
+
+import com.exalt.resourcemanagementservice.models.Request;
+import com.exalt.resourcemanagementservice.models.Server;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@Service
+public class ResourceService {
+    ArrayList<Server> servers=new ArrayList<Server>();
+    private final AtomicInteger counter = new AtomicInteger();
+
+    public Server addServer(int requestedAmount) throws InterruptedException {
+        //Server server=new Server(servers.size(),100);
+        Server server=new Server(counter.incrementAndGet(),100);
+       // server.setRequestedAmounts(requestedAmount);
+        //synchronized (server){
+        server.setRequestedAmounts(requestedAmount);
+        servers.add(server);
+        //server.setRequestedAmounts(requestedAmount);
+        Thread.sleep(5000);
```

Comment:
why did you use to sleep here? if the resources are not available, create a new one and assign it.